src/entt/entity/storage.hpp

```diff
@@ -489,8 +610,24 @@ class basic_storage: public basic_sparse_set<Entity, typename std::allocator_tra
      * @brief Returns the associated allocator.
      * @return The associated allocator.
      */
-    [[nodiscard]] constexpr allocator_type get_allocator() const noexcept {
-        return payload.get_allocator();
+    [[nodiscard]] constexpr allocator_type &get_allocator() noexcept {
```

Comment:
This is a huge no. There is no reason to return a reference to an internal allocator which isn't even required to exist.
`get_allocator` is defined as `allocator_type get_allocator() const noexcept` literally everywhere and for a good reason.

Reply:
`get_allocator` can be changed back to a return-by-value, however the internal use of `get_allocator` needs to be changed to a retrieve by reference because the allocator can be stateful and in this case a repretive call to `get_allocator` could add overhead.

Reply:
The API and the internal implementation are different beasts. I pocked more than once in the standard library implementation and I'm always amazed by the amount of... code 😅 that lies behind an API sometimes.